reader_core/src/crystal/pk2.rs

```diff
@@ -302,6 +309,16 @@ impl Pk2 {
             spc,
             hp: ((atk & 1) * 8) + ((def & 1) * 4) + ((spe & 1) * 2) + (spc & 1),
             shiny,
+            hidden_power_type: HIDDEN_POWER_LOOKUP
+                .get(((atk & 3) << 2 | def & 3) as usize)
+                .unwrap_or(&"Unknown"),
+            hidden_power_base: 31
+                + (5 * (((atk >= 8) as u8) << 3
+                    | ((def >= 8) as u8) << 2
+                    | ((spe >= 8) as u8) << 1
+                    | ((spc >= 8) as u8))
+                    + spc % 4)
+                    / 2,
```

Comment:
Two tiny nitpicks:
- Let's use bit shifts instead of casting a bool since it helps explain why we care if the DV is greater than 8 (there's a specific bit we want).
- Let's also use a bitand to grab the special DV bits for the same reason.

```suggestion
                + (5 * (((atk >> 3)) << 3
                    | ((def >> 3)) << 2
                    | ((spe >> 3)) << 1
                    | ((spc >> 3)))
                    + spc & 3)
                    / 2,
```